ir/memory.cpp

```diff
@@ -815,11 +811,29 @@ expr Pointer::blockValRefined(const Pointer &other) const {
                 val.ptrByteoffset() == val2.ptrByteoffset() &&
                 val.ptr().refined(val2.ptr());
   }
+  // allow null ptr <-> zero
+  // If ptr points to a null block with ofs, its integer address is compared.
+  // Relevant transformations:
+  //   store ty* null  <-> memset i8 0
```

Comment:
Although we allow reading 0 null, we currently don't allow reading null as 0.
Was there any transformation that required that?  If so, we need to change `bytesToValue` as well..

Reply:
I ran LLVM unit tests to check whether refinement from 0 to nullptr is needed, and it did not raise any new failure.
The motivation was also about nullptr -> 0 (https://github.com/AliveToolkit/alive2/issues/236 ), so I removed the refinement.